web-ui/src/main/resources/catalog/components/edit/onlinesrc/OnlineSrcDirective.js

```diff
@@ -723,7 +724,7 @@
                 var resetProtocol = function() {
                   scope.layers = [];
                   scope.OGCProtocol = false;
-                  if (scope.params && !scope.isEditing) {
+                  if (scope.params && !scope.isEditing && scope.clearFormOnProtocolChange) {
```

Comment:
It looks like adding a WMS layer does not work because we don't set anymore `selectedLayers`. Error returned is
![image](https://user-images.githubusercontent.com/1701393/88892118-96edd700-d244-11ea-8c69-60dfb50cc3b0.png)
so maybe only `name` and `desc` should not be modified.